README.md

```diff
@@ -1,123 +1,177 @@
+- Ease of use: Straightforward interface for maintaining xApp connections
+
+You can read more about Nomad's architecture at our [main documentation site](https://docs.nomad.xyz).
+
+## Nomad Rust Repository
+
+Nomad's off-chain systems are written in Rust for speed, safety and reliability. (Nomad's on-chain systems are written in Solidity and are available [here](https://github.com/nomad-xyz/monorepo).)
+
+### Rust Setup
+
+- Install `rustup` from [here](https://rustup.rs/) and run it
+
+Note: You should be running at least version `1.52.1` of the rustc compiler. Check it with `rustup --version`
 ```
 $ rustup --version
 rustup 1.24.2 (755e2b07e 2021-05-12)
 info: This is the version for the rustup toolchain manager, not the rustc compiler.
 info: The currently active `rustc` version is `rustc 1.52.1 (9bc8c42bb 2021-05-09)`
 ```
 
-### Useful cargo commands
+Rust uses `cargo` for package management, building, testing and other essential tasks.
 
-- `cargo doc --open`
-  - generate documentation and open it in a web browser
-- `cargo build`
-  - compile the project
-- `cargo run --example example`
-  - run the default executable for the current project
-- `cargo test`
-  - run the tests
-
-### Useful cargo extensions
-
-- tree
-  - show the dependency tree. Allows searching for specific packages
-  - install: `cargo install cargo-tree`
-  - invoke: `cargo tree`
-- clippy
-  - search the codebase for a large number of lints and bad patterns
-  - install: `rustup component add clippy`
-  - invoke: `cargo clippy`
-- expand
-  - expand macros and procedural macros. Show the code generated by the
-    preprocessor
-  - useful for debugging `#[macros]` and `macros!()`
-  - install: `cargo install cargo-expand`
-  - invoke `cargo expand path::to::module`
+For Ethereum and Celo connections we use
+[ethers-rs](https://github.com/gakonst/ethers-rs). Please see the docs
+[here](https://docs.rs/ethers/0.2.0/ethers/).
+
+Nomad uses the tokio async runtime environment. Please see the docs
+[here](https://docs.rs/tokio/1.1.0/tokio/).
+
+### Running the Test Suite
+
+- `cargo test --workspace --all-features`
+
+This will run the full suite of tests for this repository.
 
-### Architecture
+### Generate Documentation
 
-The on-chain portions of nomad are written in Solidity. The rust portions are
-exclusively off-chain. Later, there may be on-chain rust for Near/Solana/
-Polkadot.
+- `cargo doc --open`
+
+This will generate this repos documentation and open it in a web browser.
 
-Nomad will be managed by a number of small off-chain programs ("agents"). Each
-of these will have a specific role. We want these roles to be simple, and
-easily described. Each of these agents will connect to a home chain and any
-number of replicas. They need to be configured with chain connection details
-and have access to a reliable node for each chain.
+### Agent Architecture
 
-Some agent sketches:
+The off-chain portion of Nomad is a set of agents each with a specific role:
```

Comment:
Think it'd be most intuitive if order is:
1. Updater
2. Watcher
3. Relayer
4. Processor